include/lsst/afw/cameraGeom/TransformMap.h

```diff
@@ -39,30 +39,30 @@ namespace afw {
 namespace cameraGeom {
 
 /**
- * Functor for boost::transform_iterator: given an entry in a std::map or unordered_map, return the key
- */
-struct GetKey {
-    CameraSys const &operator()(std::pair<const CameraSys, int> const &p) const { return p.first; };
-};
-
-/**
  * A registry of 2-dimensional coordinate transforms for a specific camera.
  *
- * Represents the interrelationships between camera coordinate systems for a
- * particular camera. It can be seen as a mapping between a pair of CameraSys
- * and a Transform between them, but does not conform to either the C++ map or
- * Python dictionary APIs.
+ * Represents the interrelationships between camera coordinate systems for a particular camera.
+ * It can be seen as a mapping between a pair of CameraSys and a Transform between them
+ * (though it does not conform to either the C++ map or Python dictionary APIs).
  *
- * TransformMap supports transforming between any two supported CameraSys using
- * the @ref transform methods, even if no explicit transform was provided.
+ * TransformMap supports:
+ * * Transforming between any two supported CameraSys using the @ref transform methods.
+ * * Retrieving a transform between any two supported CameraSys using @ref getTransform.
+ * * Iteration over supported CameraSys using @ref begin and @ref end in C++
+ *   and standard Python iteration in Python.
  *
  * @exceptsafe Unless otherwise specified, all methods guarantee only basic
  *             exception safety.
  */
 class TransformMap final {
+private:
+     // Functor for boost::transform_iterator: given an entry in a std::map or unordered_map, return the key
+    struct GetKey {
+        CameraSys const &operator()(std::pair<const CameraSys, int> const &p) const { return p.first; };
+    };
+
```

Comment:
According to the [style guide](https://developer.lsst.io/coding/cpp_style_guide.html#the-parts-of-a-class-must-be-sorted-public-protected-and-private), the definition of `GetKey` is out of order. I don't see a way to declare but not define `CameraSysIterator`, though.

Reply:
IMO, that section of the style guide is naive, and should not apply to the rare situations (like this one) where a certain ordering (and possible repetition) of private/protected/public sections is required for code correctness.

Reply:
Just pointing out that you may catch some flak over this.

Reply:
Yeah, maybe I'll just RFC a tiny change to the style guide to add an escape clause.  I wouldn't let that RFC block this ticket.

Reply:
As per discussion on the above RFC, @ktlim, could you sign off on the above violation of C++ 5.2?

Reply:
I approve as long as `GetKey` has no general usefulness as a public interface.

Reply:
Thank you. And not `GetKey` is not used outside the class.